components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,328 @@
+///
+/// See UnicodeSet::new for conditions
+fn parse_serial_string(serialize_str: &str) -> Result<Vec<u32>, Box<dyn Error>> {
+    let mut serialize = serialize_str.split(" ");
+    let capacity: usize = serialize
+        .next()
+        .ok_or("Splitting did not yield anything")?
+        .parse()?;
+    if capacity % 2 != 0 {
+        return Err("Capacity must be even".into());
+    }
+    let mut serialized_vec: Vec<u32> = Vec::with_capacity(capacity);
+    let mut prev: Option<u32> = None;
+    for str_elem in serialize {
+        let parsed: u32 = str_elem.parse()?;
+        if serialized_vec.len() + 1 > serialized_vec.capacity() {
+            return Err("Serialization capacity is too small".into());
+        }
+        if Some(parsed) <= prev {
+            return Err("Serialization must be sorted".into());
+        }
+        serialized_vec.push(parsed);
+        prev = Some(parsed);
+    }
+    if serialized_vec.len() % 2 != 0 {
+        return Err("Serialization must be even".into());
+    }
+    serialized_vec.shrink_to_fit(); // necessary if the length < capacity
+    Ok(serialized_vec)
+}
+
+/// UnicodeSet membership wrapper
+///
+/// Provides exposure to membership functions and constructors from serialized UnicodeSets
+/// and predefined ranges.
+//#[derive(Copy, Clone, Debug, Eq)]
+pub struct UnicodeSet {
+    // If we wanted to use an array to keep the memory on the stack, there is an unsafe nightly feature
+    // https://doc.rust-lang.org/nightly/core/array/trait.FixedSizeArray.html
+    // Allows for traits of fixed size arrays
+    inv_list: Vec<u32>,
+}
+
+impl UnicodeSet {
+    /// Returns Result of UnicodeSet from serialized string
+    ///
+    /// Returns an error if the serialized string fails to parse.
+    /// The serialized string requires starting capacity integer, followed by space delimited
+    /// integer code points. There must be an even number of elements (not including the
+    /// capacity int), and must be in ascending sorted order.
+    ///
+    /// Example String: `"4 0 5 10 15"` designates a capacity of size `4`, followed by 2 ranges
+    /// The ranges are `{0, 4}` and `{10, 14}` inclusive
+    pub fn new(serialize: &str) -> Result<UnicodeSet, Box<dyn Error>> {
+        match parse_serial_string(serialize) {
+            Ok(serialize) => Ok(UnicodeSet {
+                inv_list: serialize,
+            }),
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Returns Result of UnicodeSet from a single pair of integers defining a range
+    ///
+    /// `start`: inclusive, `end`: exclusive
+    ///
+    /// Returns an error if the range is invalid (out of order and out of bounds).
+    ///
+    /// Example Call: `UnicodeSet::from_range(&0, &15)`
+    pub fn from_range(start: &u32, end: &u32) -> Result<UnicodeSet, Box<dyn Error>> {
+        if start > end {
+            return Err("Range is out of order".into());
+        }
+        if start < &CODEPOINT_MIN || end > &CODEPOINT_MAX {
```

Comment:
While I agree that this should change to the removal of references, and I will do that, the behavior is still correct:
`*s < CODEPOINT_MIN` is less confusing I agree, especially from a C++ perspective, but Rust will automatically compare the value itself when comparing references of any sort (`s <  &CODEPOINT_MIN`) unless they are specifically specified as raw, in which then it will compare memory addresses (as you would expect in C++)

Just for future reference! Thanks for the feedback